src/components/ec-date-range-filter/ec-date-range-filter.spec.js

```diff
@@ -99,4 +100,18 @@ describe('ecDateRangeFilter', () => {
     expect(wrapper.findByDataTest('ec-date-range-filter__to-input').element.value).toBe('2020-12-06');
     expect(wrapper.findByDataTest('ec-filter-popover__badge').text()).toBe('2');
   });
+
+  it('should return undefined if no dates are passed in value prop', () => {
+    const wrapper = mountecDateRangeFilterAsTemplate(
+      '<ec-date-range-filter :label="label" v-model="value"/>',
+      {},
+      {
+        data() {
+          return { value: valueEmpty, label };
+        },
+      },
+    );
+    expect(wrapper.vm.fromValueDate).toEqual(undefined);
```

Comment:
`wrapper.vm` is the vm of the Component that uses the template:
```

  const Component = localVue.extend({
    components: { ecDateRangeFilter },
    template,
    ...wrapperComponentOpts,
  });	
```
The `data` of the component only has `value` and `label`, there are no `fromValueDate`, `toValueDate` on that component, so they are always undefined regardless of what EcDataRangeFilter does. You can verify that by setting the value to something that has a value, e.g.
```
data() {
          return { value: { from: '2020-11-11' }, label };
        },
```

and the test is still green instead of failing.

Also passing `valueEmpty`, which is `{ from: null, to: null }` is not testing the new code you added: `this.value?.from;` 
`?.` means that we should test what if `this.value` is undefined. But `valueEmpty` is an object which has both properties, from and to defined.

I think all we have to test is to pass null value and check the snapshot. We don't even need the v-model for it, so we can just use
```
const wrapper = mountecDateRangeFilter({ label, value: null });
expect(wrapper.element).toMatchSnapshot();
```